Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -1488,7 +1483,49 @@ TEST_F(SerializerTest, EncodesListenRequestLabels) {
   }
 }
 
-TEST_F(SerializerTest, DecodesListenResponseWithTargetChange) {
+TEST_F(SerializerTest, DecodesMutationResult) {
+  std::vector<FieldValue> transformations({FieldValue::FromBoolean(true),
+                                           FieldValue::FromInteger(1234),
+                                           FieldValue::FromString("string")});
+  MutationResult model(Version(123456789), std::move(transformations));
+
+  v1::WriteResult proto;
+
+  proto.mutable_update_time()->set_seconds(
+      Version(123456789).timestamp().seconds());
+  proto.mutable_update_time()->set_nanos(
+      Version(123456789).timestamp().nanoseconds());
+  proto.mutable_transform_results()->Add();
+  proto.mutable_transform_results()->Add();
+  proto.mutable_transform_results()->Add();
+  (*proto.mutable_transform_results())[0] = ValueProto(true);
```

Comment:
Nit: I'd move each assignment to be right after the `Add`.

Reply:
Done.

Reply:
`Add` returns a pointer to the thing that was added. You could avoid manual indexing with something like this:

```c++
auto transform_results = proto.mutable_transform_results();
*transform_results->Add() = ValueProto(true);
*transform_results->Add() = ValueProto(1234);
*transform_results->Add() = ValueProto("string");
```

(Note this works because member access (`->` or `.`) has higher precedence than indirection (`*`).

Reply:
Done, thanks!